kapre/augmentation.py

```diff
@@ -0,0 +1,85 @@
+import tensorflow as tf
+from tensorflow.keras import backend as K
+from . import backend
+from .backend import _CH_FIRST_STR, _CH_LAST_STR, _CH_DEFAULT_STR
+from tensorflow.python.ops import array_ops
+from tensorflow.keras.layers import Layer
+from tensorflow.python.keras.utils import tf_utils
+
+__all__ = [
+    'SpecAugment'
+]
+
+class SpecAugment(Layer):
+    """
+    # `SpecAugment`
+
+    ```python
+    kapre.augmentation.SpecAugment(time_param=10, freq_param=10, **kwargs)
+    ```
+
+    Add masking to input data and output it
+    Args:
+        x: Spectrogram
+        freq_param: Param of freq masking
+        time_param: Param of Time masking
+    Returns: 
+        Masked Tensor of Spectrogram
+    """
+    def __init__(
+        self, 
+        freq_param = None, 
+        time_param = None,
+        input_data_format:str='default',
+        **kwargs
+    ):
+        
+        super(SpecAugment, self).__init__(**kwargs)
+
+        backend.validate_data_format_str(input_data_format)
+
+        self.freq_param = freq_param
+        self.time_param = time_param
+        self.uses_learning_phase = True
+        self.supports_masking = True
+
+        assert (freq_param is not None) or (time_param is not None), "at least one param value should be defined"
+        
+        idt = input_data_format
+        self.input_data_format = K.image_data_format() if idt == _CH_DEFAULT_STR else idt
+
+    def call(self, x, training=None):
+        
+        if training is None:
+            training = K.learning_phase()
+            
+        if self.input_data_format == 'channels_first':
+            assert x.shape[1] == 1, 'SpecAugment does not support 2D images yet'
```

Comment:
Can't we just extend it so that `SpecAug` is applied to all the channels? I'd like `Kapre` to be as versatile as possible.

Reply:
Btw in this kind of code, I'd prefer
```
ch_axis = 1 if self.input_data_format == 'channels_first' else 3
if K.int_shape(x)[ch_axis] != 1:
    raise RuntimeError('error msg, why it is broken, what was the expected value, what value it received, etc')
```

Reply:
Well, I think I didn't describe the error correctly, It does support 2D images what It doesn't support is masking for images with depth >  1, but I'll work on that as soon as I have available time

Reply:
yeah but what's the reason behind the limitation? i think as a package, if kapre has a specaug, it should definitely support multi-channel inputs.